### ISSUE TITLE
Spontaneous rotation occurs right before the molecule breaks

### ISSUE THREAD

**Issue by bryo5363** [phetsims/molecules-and-light#110]
Test device: MacBook Pro

Operating System: 10.10.1

Browser: Firefox 36.0.4

Problem description: The moment before a molecule breaks, there is a spontaneous rotation to it's original state

Steps to reproduce:
-open sim
-select molecule with two bonds
-select microwave so that the molecule rotates
-once the molecule has deviated from its original position, hit it w/ ultraviolet
-notice the molecule returns to its initial position before breaking
Severity:

Screenshots:
![screen shot 2015-04-02 at 10 26 56 am](https://cloud.githubusercontent.com/assets/7717368/6968382/a8668d9e-d923-11e4-8be6-76d2502d0c75.png)
![screen shot 2015-04-02 at 10 27 11 am](https://cloud.githubusercontent.com/assets/7717368/6968383/ac84f244-d923-11e4-91a0-dd6e382d8a2f.png)

Troubleshooting information (do not edit):
Name: Molecules and Light
URL: http://www.colorado.edu/physics/phet/dev/html/molecules-and-light/1.0.0-rc.1/molecules-and-light_en.html
Version: 1.0.0-rc.1
Features missing: touch
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.104 Safari/537.36
Language: en-US
Window: 1210x567
Pixel Ratio: 1/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 15 uniform: 1024
Texture: size: 8192 imageUnits: 16 (vertex: 16, combined: 16)
Max viewport: 8192x8192
OES_texture_float: true

**Comment by jessegreenberg** [comment]
Good catch @bryo5363, and it looks like this behavior is present in the Java version as well.

**Comment by jessegreenberg** [comment]
Looking through the code, it looks like this behavior is an explicit design choice.  Constituent molecules are set so that their initial positions match the initial layout of the dissociated molecule.

@jbphet, do you know why this behavior was chosen rather than setting the constituent molecule positions to match the current orientation of the initial molecule?

For quick reference, here is where this behavior is defined in the break apart strategy ( line 139 of NO2.js ):
```
// Set up the direction and velocity of the constituent molecules.  These are set up mostly to look good, and
// their directions and velocities have little if anything to do with any physical rules of atomic dissociation.
      var diatomicMoleculeRotationAngle = ( ( Math.PI / 2 ) - ( INITIAL_OXYGEN_NITROGEN_OXYGEN_ANGLE / 2 ) );
      var breakApartAngle;
      if ( this.doubleBondOnRight ) {
        nitrogenMonoxideMolecule.rotate( -diatomicMoleculeRotationAngle );
        nitrogenMonoxideMolecule.setCenterOfGravityPos( ( this.getInitialAtomCogOffset( this.nitrogenAtom ).x + this.getInitialAtomCogOffset( this.rightOxygenAtom ).x ) / 2,
          ( this.getInitialAtomCogOffset( this.nitrogenAtom ).y + this.getInitialAtomCogOffset( this.rightOxygenAtom ).y ) / 2 );
        breakApartAngle = Math.PI / 4 + Math.random() * Math.PI / 4;
        singleOxygenMolecule.setCenterOfGravityPos( -this.initialOxygenHorizontalOffset, this.initialOxygenVerticalOffset );
      }
      else {
        nitrogenMonoxideMolecule.rotate( Math.PI + diatomicMoleculeRotationAngle );
        breakApartAngle = Math.PI / 2 + Math.random() * Math.PI / 4;
        nitrogenMonoxideMolecule.setCenterOfGravityPos( ( this.getInitialAtomCogOffset( this.nitrogenAtom ).x + this.getInitialAtomCogOffset( this.leftOxygenAtom ).x ) / 2,
          ( this.getInitialAtomCogOffset( this.nitrogenAtom ).y + this.getInitialAtomCogOffset( this.leftOxygenAtom ).y ) / 2 );
        singleOxygenMolecule.setCenterOfGravityPos( this.initialOxygenHorizontalOffset, this.initialOxygenVerticalOffset );
      }
```
Very similar behavior defined for O3 ( on line 126 ).

**Comment by jbphet** [comment]
I'm fairly certain that this was a conscious design choice intended to give us greater control over the visual appearance of the break apart events and to minimize the complexities of interaction between two motion strategies, in this case the rotation strategy and the break apart strategy.  If I remember it correctly, the designers wanted the molecule fragments to always go off the top and bottom of the screen.  Also, implementation was more difficult if we wanted to be able to support breaking apart from any arbitrary rotational position.  Ergo, rotate back to original position just before breaking apart.

I tested this scenario several times on v1.0.0-rc.1, and in my opinion the rotation is not really noticeable.  You pretty much have to pause the sim and step the sim in order to see this behavior, which I think very few users would ever do.

I discussed this issue with @ariel-phet and, since the behavior was intentional, and since we've never had a complaint about it on the Java sim, we are planning to leave this as is.  I will put a note about it in the implementation notes and improve the comments in the code.
